src/robot/libdocpkg/jsonbuilder.py

```diff
@@ -0,0 +1,81 @@
+#  Copyright 2020-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import os.path
+
+from robot.errors import DataError
+from robot.running.arguments import ArgumentSpec, ArgInfo
+
+from .model import LibraryDoc, KeywordDoc
+
+
+class JsonDocBuilder(object):
+
+    def build(self, path):
+        spec = self._parse_spec_json(path)
+        return self.build_from_dict(spec)
+
+    def build_from_dict(self, libdoc_dict):
+        libdoc = LibraryDoc(name=libdoc_dict.get('name'),
+                            doc=libdoc_dict.get('doc'),
+                            version=libdoc_dict.get('version'),
+                            type=libdoc_dict.get('type'),
+                            scope=libdoc_dict.get('scope'),
+                            doc_format=libdoc_dict.get('doc_format'),
+                            source=libdoc_dict.get('source'),
+                            lineno=int(libdoc_dict.get('lineno')))
```

Comment:
- Does `libdoc_dict` have extra items or only those that are accessed? If not, all this could be replaced with `libdoc = LibraryDoc(**libdoc_dict)`.

- Above proposal wouldn't convert `lineno` to `int` but I would assume it's `int` in JSON as well. I think it may also be `null` in which case `int´ conversion would actually fail.

- `libdoc_dict` might be better named `spec` like it's i `build`. Also `build_from_dict´ could be `build_from_spec`.

- Is the idea that `build_from_dict` is public? If not, prefix with `_` .

Reply:
@pekkaklarck 
1. in the json are more information than the constructor accepts.
example: Generated, all_tags, keywords and inits

2. changed the int to int(spec.get('lineno', -1)) as in specbuilder.

3. and 4. Yea the idea was that that could maybe used to generate Libdoc from the same format that JSON creates but as dict, because the `build` method only accepts a path. so if someone/something want to use the jsonbuilder it would need to write a file first.
I did change the variable name to `spec` but i wouldn’t change the method name, because it points out that compared to `build` from a path that one can `build` the model from a dictionary.

i changed all `.get('key')` to normal dictionary syntax